R/TuningInstanceSingleCrit.R

```diff
@@ -115,10 +115,24 @@ TuningInstanceSingleCrit = R6Class("TuningInstanceSingleCrit",
     #' This defines the resampled performance of a learner on a task, a
     #' feasibility region for the parameters the tuner is supposed to optimize,
     #' and a termination criterion.
-    initialize = function(task, learner, resampling, measure, search_space,
+    initialize = function(task, learner, resampling, measure, search_space = NULL,
       terminator, store_benchmark_result = TRUE, store_models = FALSE,
       check_values = FALSE) {
         measure = as_measure(measure)
+        # We might want to have the following in a function bc its used in MultiCrit as well
+        if (is.null(search_space)) {
+          # TODO: check if we can construct search space from learner$param_set using tune_tokens
+          tmp = learner$param_set$get_tune_pair() # return fixed_values (all but tune tokens) and search_space (from tune tokens),
```

Comment:
I don't think we should have a function that returns two things that easily could be obtained by two calls, since here the two things that would happen are independent of each other.

Reply:
Just to emphasize that you always want to do both? If you want to have the search_space you also always should take the param_set with the remaining fixed param values.